cmake/modules/snippets.cmake

```diff
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2021, Nordic Semiconductor ASA
+
+# Snippets support
+#
+# Outcome:
+# The following variables will be defined when this module completes:
+#
+# - SNIPPET_AS_LIST: CMake list of snippet names, created from the
+#   SNIPPET variable
+
+include_guard(GLOBAL)
+
+include(extensions)
+
+# Warn the user if SNIPPET changes later. Such changes are ignored.
+zephyr_check_cache(SNIPPET WATCH)
+
+# Putting the body into a function prevents us from polluting the
+# parent scope. We'll set our outcome variables in the parent scope of
+# the function to ensure the outcome of the module.
+function(zephyr_process_snippets)
+  if (SNIPPET)
```

Comment:
minor detail, this will prevent the use of snippets called `false`, `no`, `n`, and similar.
Probably a minor thing, but for this `if (DEFINED SNIPPET)` is probably a better choice.